Moved ASP.NET Core templates and logic to separated AddIn

DavidKarlas opened the pull request.

mrward commented on main/src/addins/MonoDevelop.AspNetCore/Properties/AddinInfo.cs:
```diff
@@ -0,0 +1,16 @@
+                 Version = MonoDevelop.BuildInfo.Version,
+                 Category = "Web Development")]
+
+[assembly:AddinName (".NET Core Support")]
```
ASP.NET Core Support?

mrward replied:
Should we also move the ASP.NET Core specific code from the [DotNetCoreRunConfiguration class](https://github.com/mono/monodevelop/blob/master/main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreRunConfiguration.cs) and move it into the AspNetCore addin? We could create an AspNetCoreRunConfiguration class that maybe derives from DotNetCoreRunConfiguration.

Could we have an AspNetCoreProjectExtension that is available for project's that have the AspNetCore project capability?

mrward replied:
Also maybe some parts of the DotNetCoreExecutionHandler should be moved across into an AspNetCoreExecutionHandler? There is code in there for launching the browser.

mrward replied:
Also maybe the ASP.NET Core specific parts of the DotNetCoreRunConfigurationEditorWidget should be split out and moved into the AspNetCore addin? Although maybe the whole thing should go across since it looks like if the project is not a web project there is no extra UI added.

DavidKarlas replied:
Last time I tried splitting DotNetCoreRunConfiguration into 2 was problem with breaking API :( I will take another look

DavidKarlas commented:
@mrward addressed comments and converted more code to AspNetCore, unfortunately for backward compatibility reasons, I didn't delete code in DotNetCore addin, this can be done when we bump to VSfM 8.0 version

mrward commented on main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore/AspNetCoreProjectExtension.cs:
```diff
@@ -0,0 +1,83 @@
+namespace MonoDevelop.AspNetCore
+{
+	[ExportProjectModelExtension]
+	public class AspNetCoreProjectExtension : DotNetCoreProjectExtension
```
This probably should not be public.

mrward commented on main/src/addins/MonoDevelop.DotNetCore/MonoDevelop.DotNetCore/DotNetCoreRunConfigurationEditor.cs:
```diff
@@ -51,116 +51,26 @@ public override Control CreateControl ()
 		}
 	}
 
 	class DotNetCoreRunConfigurationEditorWidget : DotNetRunConfigurationEditorWidget
```
Can this class be deleted now? Does not seem to be used.

DavidKarlas replied:
No because it still has function, that is call `DotNetRunConfigurationEditorWidget` with `false` in constructor which results in Mono runtime selection pad removed

DavidKarlas commented:
Removed public from project extension class

mrward commented on main/src/addins/MonoDevelop.AspNetCore/MonoDevelop.AspNetCore/DotNetCoreExecutionHandler.cs:
```diff
@@ -0,0 +1,107 @@
+
+namespace MonoDevelop.AspNetCore
+{
+	class AspNetCoreExecutionHandler : IExecutionHandler
```
Filename is DotNetCoreExecutionHandler.cs but the class name is AspNetCoreExecutionHandler.

DavidKarlas replied:
Fixed
